composer/algorithms/alibi/README.md

```diff
@@ -0,0 +1,108 @@
+# ⚖️ ALiBi
+
+[\[How to Use\]](#how-to-use) - [\[Suggested Hyperparameters\]](#suggested-hyperparameters) - [\[Technical Details\]](#technical-details) - [\[Attribution\]](#attribution)
+
+ `Natural Language Processing`
+
+ALiBi (Attention with Linear Biases) dispenses with position embeddings for tokens in transformer-based NLP models, instead encoding position information by biasing the query-key attention scores proportionally to each token pair’s distance. ALiBi yields excellent extrapolation to unseen sequence lengths compared to other position embedding schemes. We leverage this extrapolation capability by training with shorter sequence lengths, which reduces the memory and computation load.
+
+| ![Alibi](https://storage.googleapis.com/docs.mosaicml.com/images/methods/alibi.png) |
+|:--:
+|*The matrix on the left depicts the attention score for each key-query token pair. The matrix on the right depicts the distance between each query-key token pair. m is a head-specific scalar that is fixed during training. Figure from [Press et al., 2021](https://openreview.net/forum?id=R8sQPpGCv0).*|
+
+## How to Use
+
+### Functional Interface
+
+```python
```

Comment:
@ajaysaini725 - Please add a bit of text at the beginning explaining what's happening.